crates/net/eth-wire/src/p2pstream.rs

```diff
@@ -519,14 +521,16 @@ pub fn set_capability_offsets(
     let mut shared_capability_names = BTreeSet::new();
 
     // find highest shared version of each shared capability
-    for capability in peer_capabilities {
+    for peer_capability in peer_capabilities {
         // if this is Some, we share this capability
-        if let Some(version) = our_capabilities_map.get(&capability.name) {
+        if our_capabilities.get(&peer_capability).is_some() {
             // If multiple versions are shared of the same (equal name) capability, the numerically
             // highest wins, others are ignored
-            if capability.version <= *version {
```

Comment:
What was the bug here? And why do we query the shared caps now vs not before?

Reply:
This code assumes that the local node supports all lower versions of the given capability.

Also, the logic for "support the highest shared version of the capability" here only works under the assumption that we iterate over the peers capabilities in order of increasing version.

We now query the shared capabilities because that's where we keep track of the highest shared version of a capability that we've seen yet